File: src/bin/utf8_support.cpp

#if defined(_WIN32) || defined(_WIN64)
#include <Windows.h>
#endif

#include <filesystem>
#include <iostream>

int main() {
#if defined(_WIN32) || defined(_WIN64)
    SetConsoleOutputCP(CP_UTF8);
#endif
    // This command is required to print UTF-8 characters to the console.
    // But in many other languages, utf8 is the default encoding.

    auto current_dir = std::filesystem::current_path() / "data";
    for (auto &entry : std::filesystem::directory_iterator(current_dir)) {
        auto path = entry.path().filename().string();
        std::cout << "Filename: " << path << std::endl;
    }
}

File: src/bin/utf8_support.rs
use anyhow::{anyhow, Result};
fn main() -> Result<()> {
    let current_dir = std::env::current_dir()?.join("data");
    // read dir
    let read_dir = std::fs::read_dir(current_dir)?;
    for entry in read_dir {
        let file_name = entry?.file_name();
        let file_name = file_name
            .to_str()
            .ok_or(anyhow!("Invalid UTF-8 sequence"))?;
        println!("Filename: {}", file_name);
    }
    Ok(())
}
